docs/markdown/Python/python-goals/python-test-goal.md

```diff
@@ -67,9 +67,28 @@ python_tests(
 )
 ```
 
-> 🚧 Avoid the `pytest-xdist` plugin
+> 🚧 How to use the `pytest-xdist` plugin
 >
-> We do not recommend using this plugin because its concurrency conflicts with Pants' own parallelism. Using Pants will bring you similar benefits to `pytest-xdist` already: Pants will run each test target in parallel.
+> Pants' default concurrency will run each test target in parallel. If you want to additionally parallelize _within_  each of your targets, you can do so by enabling built-in support for the `pytest-xdist` plugin:
+>
+> ```toml pants.toml
+> [pytest]
+> xdist_enabled = True
+> ```
+>
+> This will cause Pants to pass `-n <concurrency>` when running `pytest`.
+>
+> By default, Pants will automatically compute the value of `<concurrency>` for each target based on the number of tests defined in the file and the number of available worker threads. You can instead set a hard-coded concurrency value per target:
+>
+> ```python BUILD
+> python_test(name="tests", source="tests.py", xdist_concurrency=4)
+> ```
+>
+> To explicitly disable the use of `pytest-xdist` for a target, set `xdist_concurrency=0`.
```

Comment:
Maybe motivate this?

```suggestion
> To explicitly disable the use of `pytest-xdist` for a target, set `xdist_concurrency=0`. This can be necessary for tests that are not safe to run in parallel.
```